adagrams/game.py

```
@@ -1,11 +1,130 @@
+    "L": 1,
+    "M": 3,
+    "N": 1,
+    "O": 1,
+    "P": 3,
+    "Q": 10,
+    "R": 1,
+    "S": 1,
+    "T": 1,
+    "U": 1,
+    "V": 4,
+    "W": 4,
+    "X": 8,
+    "Y": 4,
+    "Z": 10
+}
+
+def make_letter_list(dict):
+    #Turn dict into list where each letter appears a number of times
+    #equal to its frequency
+    letter_list = []
+    for letter, count in dict.items():
+        letter_list.extend([letter for _ in range(count)])
+
+    return letter_list
+
+
 def draw_letters():
-    pass
+    letter_list = make_letter_list(LETTER_POOL)
+
+    #Create an empty hand, then add 10 random letters from letter list.
+    #Letters are removed from list as they are added to hand.
+    hand = []
+
+    for _ in range(10):
+        chosen_letter = r.choice(letter_list)
+        hand.append(chosen_letter)
+        letter_list.remove(chosen_letter)
+
+    return hand
+
+def word_to_dict(word):
+    word_dict = {}
+    for i in range(len(word)):
+        letter = word[i].upper()
+        word_dict[letter] = word_dict.get(letter, 0) + 1
+
+    return word_dict
 
 def uses_available_letters(word, letter_bank):
-    pass
+    #Create dicts of hand and current word
+    letter_dict = word_to_dict("".join(letter_bank))
+    word_dict = word_to_dict(word)
+
+    #Compare to verify word letters are in bank AND there aren't too many
+    for letter, count in word_dict.items():
+        if letter not in letter_dict or count > letter_dict[letter]:
+            return False
+
+    return True
 
 def score_word(word):
-    pass
+    #Iterate through word and total score for each letter
+    score = sum([SCORE_DICT[word[i].upper()] for i in range(len(word))])
+
+    #Add 8 extra points if word is 7-10 letters long
+    if len(word) >= 7:
+        score += 8
+
+    return score
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    #Set initial score tuple to ("word", -1) so first word will always overwrite
+    high_score = ("word", -1)
```

Setting an "impossible" default is a frequently used strategy for doing something like this. Another can be to assume the first value in a list is the winner until disproven (sometimes results in a self-comparison, but this can b avoided if it becomes a problem).

Consider still having an explicit empty  list check (maybe there was aa round of the game in which no one was able to make a word?). We could return `None`, which is a slightly more expected sentinel. Or we could raise an error (`max` effectively takes this approach, deciding that there's no such think as a max in an empty list and asking for one indicates we may have done something wrong. N.B. max does allow for a default if we prefer that behavior as well).